app/models/concerns/softly_deletable.rb

```diff
@@ -8,7 +8,9 @@ module SoftlyDeletable
     scope :not_deleted, -> { where(deleted_at: nil) }
   end
 
-  def soft_delete!
+  def soft_delete!(reason_code = nil, comment = nil)
+    Reason.create(ad: self, code: reason_code, comment: comment) if reason_code || comment
```

Comment:
Should we use `Reason.create!` here so it raises error if reason is passed but could not be saved?
```suggestion
    Reason.create!(ad: self, code: reason_code, comment: comment) if reason_code || comment
```

Reply:
I thought about that but opted to not do it since reason is collected for internal information so I didn't want to block the user in this case. Wdyt?

Reply:
Agreed!